src/RpcServer/RpcServer.cs

```diff
@@ -0,0 +1,243 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.ResponseCompression;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+using Microsoft.Extensions.DependencyInjection;
+using Neo.IO;
+using Neo.IO.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Security;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Neo.Plugins
+{
+    public sealed partial class RpcServer : Plugin
+    {
+        private static readonly Dictionary<string, Func<JArray, JObject>> methods = new Dictionary<string, Func<JArray, JObject>>();
+        private IWebHost host;
+
+        public RpcServer()
+        {
+            RegisterMethods(this);
+        }
+
+        private bool CheckAuth(HttpContext context)
+        {
+            if (string.IsNullOrEmpty(Settings.Default.RpcUser)) return true;
+
+            context.Response.Headers["WWW-Authenticate"] = "Basic realm=\"Restricted\"";
+
+            string reqauth = context.Request.Headers["Authorization"];
+            string authstring;
+            try
+            {
+                authstring = Encoding.UTF8.GetString(Convert.FromBase64String(reqauth.Replace("Basic ", "").Trim()));
+            }
+            catch
+            {
+                return false;
+            }
+
+            string[] authvalues = authstring.Split(new string[] { ":" }, StringSplitOptions.RemoveEmptyEntries);
+            if (authvalues.Length < 2)
+                return false;
+
+            return authvalues[0] == Settings.Default.RpcUser && authvalues[1] == Settings.Default.RpcPass;
```

Comment:
A conflict of namespace here about ```Settings``` on ApplicationLogs and RpcServer which is able to return false here.

Reply:
```
authvalues[0]
"aaa"
Settings.Default.RpcUser
error CS0433: The type 'Settings' exists in both 'ApplicationLogs, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null' and 'RpcServer, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'
```
![image](https://user-images.githubusercontent.com/38493437/70590107-a57f7380-1c0c-11ea-8b3b-3860ed62c8ba.png)

Reply:
No problem here. They are both `internal`.

Reply:
ApplicationLogs is using RpcServer as references. Is it related to this? Loading sequence issue?
![image](https://user-images.githubusercontent.com/38493437/70596586-411adf00-1c21-11ea-9db0-240a57ccad21.png)